### ISSUE TITLE
OpenSyobonAction's license

### ISSUE THREAD

**Issue by TheZeus121** [angelXwind/OpenSyobonAction#6]
There is no license specified for the source code and the resources.  The following is not enough for this to qualify as an open-source project, since the following doesn't even allow modification and isn't clear on redistribution ("if you wish" doesn't mean "you can").

> * If any problems are caused by this game, please understand that the developers cannot take any responsibility.
> * If you wish to distribute this game to a large amount of people, please give credit to the original developer, Chiku (ちく).

----------------------------------------

How to fix this:

* Add a free license such as [GPL-3.0-or-later](https://www.gnu.org/licenses/gpl-3.0.en.html) (recommended) or [Apache-2.0](https://www.apache.org/licenses/LICENSE-2.0.html) to the source code.
* License the resource files under a free license, such as [CC-BY-SA-4.0](https://creativecommons.org/licenses/by-sa/4.0/).

This is of course only possible if you have created all of the files, which I assume might not be the case since this is a port, in which case you need to get a permission from the original author(s) to re-license their work.

**Comment by ghost** [comment]
Yes, the project needs a license, I would recommend BSD 2-Clause or MIT, as they are the true definition of "free" because they do not have many requirements.

**Comment by akemin-dayo** [comment]
Hi everyone — sorry for the literal years-late reply.

While I _do_ agree that the project should ideally be relicensed to one of the various established open-source licenses (preferably a permissive one such as MIT or BSD 2-clause, which are usually my go-tos), this is unfortunately a rather complicated situation.

---

This is the original "license" that was included with the original Syobon Action source code from 2007, in Japanese:
```
スパゲティソース かつ タブが一度も使用されてないうえ、
クラスわけ無しにif文大量使用のカオスを解読できる方はどーぞｗ
ご自由に改造なさって下さい。

ただ、そのままのソースを配布するのはご遠慮下さい。
```

Here's my English translation of the above:
```
If you can somehow understand this spaghetti code made up of if statements with
no class separation and no indentation whatsoever, please feel free to modify
it as you wish.

However, please refrain from redistributing the unmodified source code as-is.
```

(※ I am aware that this repo does not have a complete history of all changes prior to RC 2 tracked in Git — this will be corrected.)

---

All that being said, in order to _relicense_ the code and associated assets, you must seek permission from _all_ other contributors for a given project.

Unfortunately, my attempts at getting into contact with the other upstream contributors several years ago in 2019 (when this issue was first created) have failed — I never got a response from anyone involved.

(In fact, that's why this reply is so delayed — I thought I would post a comment here after getting a reply but uh… that sure worked out!)

As a result, I unfortunately do not think I can legally relicense this.

I'll keep this issue thread open, though, as I do want to perhaps make another attempt sometime again now that some time has passed (so as to not be a bother).

※ Also, I plan to fully clean up this project in general, including doing another translation pass of the README from the original Japanese text so that it sounds a little less awkward in English.

**Comment by swagtoy** [comment]
If the content wasn't licensed initially then it is considered 
public domain

**Comment by akemin-dayo** [comment]
@nekobbbbbbit I am (obviously) not a lawyer, but uh… I don't think that's correct, based on everything else I've read about similar situations.

I'll try again to establish contact sometime soon and hope for a proper answer.

**Comment by swagtoy** [comment]
> @nekobbbbbbit I am (obviously) not a lawyer, but uh… I don't think that's correct, based on everything else I've read about similar situations.
> 
> I'll try again to establish contact sometime soon and hope for a proper answer.

did you write the source code?

https://sourceforge.net/projects/opensyobon/ specifies it as GPLv2

**Comment by akemin-dayo** [comment]
…

Gods, all these years and I never saw that. Thank you for sharing your findings.

It's odd that a `LICENSE` file was never included in either the source tarball or the binary distributions, though — fairly certain that's required by GPLv2.

I'll still make another attempt at contacting both upstream contributors just to be _extra_ sure — notably, I want to make sure that the GPLv2 relicense was done with the permission of the original author.

If I don't receive a reply, I will just go ahead with licensing this repo as GPLv2.

**Comment by swagtoy** [comment]
Please ping me for an update. I submitted a port email to ports@openbsd.org and it's ready. I even patched `pledge(2)` support, but only thing holding it back is the license

**Comment by swagtoy** [comment]
@akemin-dayo any update on this?

**Comment by Olf0** [comment]
@akemin-dayo, WRT the [GPLv2-licensed](https://sourceforge.net/projects/opensyobon/#top_nav) `opensyobon` source code of [v0.9, RC1 and RC2 at SourceForge](https://sourceforge.net/projects/opensyobon/files/src/), please also mind this issue report there: "[This is … Syobon Action 2 (an unofficial fan sequel), …](https://sourceforge.net/p/opensyobon/bugs/4/)"

Side notes:
- The user `twoscomplement` who uploaded these three source code drops (v0.9 on 2010-05-23, RC1 on 2010-07-01, RC2 on 2010-09-15) calls himself [Mathew Velasquez](https://sourceforge.net/u/twoscomplement/profile/) and seems to have used this SourceForce account solely for creating the `opensyobon` project page and uploading these three source archives together with their game data archives as well as their [corresponding binaries](https://sourceforge.net/projects/opensyobon/files/).
- The user `SomeOne` who created aforementioned [issue #4 "Inaccurate Project Description"](https://sourceforge.net/p/opensyobon/bugs/4/) at SourceForge calls her-/himself [catmariopro](https://sourceforge.net/u/catmariopro/profile/) and used this SourceForce account [solely for this single issue report](https://sourceforge.net/u/catmariopro/activity/?page=0&limit=100)!

I do not remember, if one can contact other SourceForge users when being logged in and I seem to have forgotten my credentials for SourceForge.

Note that I also have not downloaded the [source code archive](https://sourceforge.net/projects/opensyobon/files/src/SyobonAction_rc2_src.tar.gz/download) and [data archive](https://sourceforge.net/projects/opensyobon/files/src/SyobonAction_rc2_data.tar.gz/download) of Syobon Action RC2 from SourceForge and compared it with a [Git-checkout of commit 5902a6b here](https://github.com/akemin-dayo/OpenSyobonAction/tree/5902a6b0cb0ab9e6771e61a41b831db6f2833194) to see if they are identical.

------------------------------------

P.S.: The [Syobon Action Fandom Wiki](https://syobon-action.fandom.com/wiki/Special:AllPages) and the [English Wikipedia](https://en.wikipedia.org/wiki/Shobon_no_Action) does not seem to reveal any additional information of the source code history, maybe the [Japanese Wikipedia](https://ja.wikipedia.org/wiki/%E3%82%A8%E3%83%AA%E3%82%A3%E3%81%AE%E3%82%A2%E3%82%AF%E3%82%B7%E3%83%A7%E3%83%B3#%E3%81%97%E3%82%87%E3%81%BC%E3%82%93%E3%81%AE%E3%82%A2%E3%82%AF%E3%82%B7%E3%83%A7%E3%83%B3) does?

**Comment by M0REKZ** [comment]
Btw the fact that this is based in "Syobon Action 2" is not wrong, there is even part of the readme of that version in this repo https://github.com/akemin-dayo/OpenSyobonAction/blob/master/README_ja.md#%E9%96%8B%E7%99%BA%E8%80%85%E3%81%AE%E3%83%A1%E3%83%A2

(compare with https://yaruki0.net/programs/action.html

and

https://web.archive.org/web/20251203070738*/http://www.geocities.jp/z_gundam_tanosii/home/syobon-action-src.zip)

So I guess with this there are at least 3 developers that touched this code (Chiku, Bluvel (alternative name for 大川和樹) and Mathew Velasquez?)
